src/lib/Calendar.js

```diff
@@ -1,522 +1,289 @@
-          _.eq( candidate.type, Types.FEAST )
-          && ( _.eq( date.type, Types.FERIA ) || _.eq( date.type, Types.SUNDAY ) )
-          && !_.eq( date.data.season.key, LiturgicalSeasons.LENT )
-          && candidate.data.prioritized
-        ) {
-          replace = true;
-        }
+  /**
+   * Push new DateItem objects in the Calendar object
+   */
+  _push(sources: (Object[])[]) {
+    sources
+      .forEach((cal, i) => cal
+        .forEach((itemObj) => {
+          this._checkAndRemoveExistingItem(itemObj);
+
+          // Specify the stack index of the source calendar
+          itemObj._stack = i;
+
+          // Get the base DateItem to attach on the new DateItem
+          let baseItem = this._find({date: itemObj.moment.toISOString(), _stack: 0});
+
+          // Create a new DateItem and add it to the collection
+          let item = new DateItem(itemObj, baseItem);
+          this.dateItems.push(item);
+        }));
+  }
 
-        //------------------------------------------------------------------
-        // A Sunday can only replace a Sunday
-        //------------------------------------------------------------------
-        else if (
-          _.eq( date.type, Types.SUNDAY )
-          && _.eq( candidate.type, Types.SUNDAY )
-        ) {
-          replace = true;
+  /**
+   * If a previous date item already exists (have a same key name), it will be removed in favour of
+   * the new given one, except if the previous item is prioritized but not the new one.
+   */
+  _checkAndRemoveExistingItem(item) {
+    // If a previous date item already exists (have a same key name), it will be removed in favour of
+    // the new one, except if the previous item is prioritized but not the new one.
+    let previousItems = this._filter({key: item.key});
+    if (previousItems.length) {
+      previousItems.forEach((previousItem) => {
+        if ((!previousItem.data.prioritized) ||
+          (previousItem.data.prioritized && item.data && item.data.prioritized)) {
+          this._removeWhere({_id: previousItem._id});
         }
+      });
+    }
+  }
 
-        //------------------------------------------------------------------
-        // A feria can only replace a feria
-        // Sundays in Ordinary Time and Chrismastide take precedence over all other celebrations
-        // except for SOLEMNITIES and certain FEASTS
-        //------------------------------------------------------------------
-        else if (
-          _.eq( date.type, Types.FERIA )
-          && _.eq( candidate.type, Types.FERIA )
-        ) {
-          replace = true;
+  /**
+   * Sort all DateItems by relevance (more relevant first)
+   * and drop non relevant DateItems
+   */
+  _sortAndKeepRelevant() {
+
+    // Reorder the type ranking, so when there is only optional items (in addition to the FERIA),
+    // the FERIA item is moved first before the optional items,
+    // since it's the default item if none of the optional items are used.
```

Comment:
Are you suspicious? optional items mean optional memories and commemorations 😉 

Actually, they are ordered by last defined first. For example, a celebration defined in the national calendar will be before of the one defined in the general calendar (but still after the feria). Depending on the localization it might be difficult to sort them alphabetically. Also it's delicate to modify things here, because this part of code is critical for all the celebrations management.

Reply:
Oh, sorry, I’ve forgotten about the commemlrations. So _optionl ktems_ are good enough. :smiley:

As for the sorting: just keep it the way it is. :smiley: